sensor/common/image/service_impl_test.go

```diff
@@ -0,0 +1,239 @@
+package image
+
+import (
+	"context"
+	"testing"
+
+	"github.com/golang/mock/gomock"
+	"github.com/pkg/errors"
+	v1 "github.com/stackrox/rox/generated/api/v1"
+	"github.com/stackrox/rox/generated/internalapi/sensor"
+	"github.com/stackrox/rox/generated/storage"
+	"github.com/stackrox/rox/pkg/concurrency"
+	cacheMocks "github.com/stackrox/rox/pkg/expiringcache/mocks"
+	"github.com/stackrox/rox/sensor/common"
+	imageMocks "github.com/stackrox/rox/sensor/common/image/mocks"
+	"github.com/stretchr/testify/suite"
+)
+
+func TestImageService(t *testing.T) {
+	suite.Run(t, new(imageServiceSuite))
+}
+
+type imageServiceSuite struct {
+	suite.Suite
+	mockCtrl          *gomock.Controller
+	mockCache         *cacheMocks.MockCache
+	mockRegistryStore *imageMocks.MockregistryStore
+	mockCentral       *imageMocks.MockcentralClient
+	mockLocalScan     *imageMocks.MocklocalScan
+	service           *serviceImpl
+}
+
+var _ suite.SetupTestSuite = (*imageServiceSuite)(nil)
+
+func (s *imageServiceSuite) SetupTest() {
+	s.mockCtrl = gomock.NewController(s.T())
+	s.mockCache = cacheMocks.NewMockCache(s.mockCtrl)
+	s.mockRegistryStore = imageMocks.NewMockregistryStore(s.mockCtrl)
+	s.mockCentral = imageMocks.NewMockcentralClient(s.mockCtrl)
+	s.mockLocalScan = imageMocks.NewMocklocalScan(s.mockCtrl)
+}
+
+func (s *imageServiceSuite) createImageService() {
+	s.service = &serviceImpl{
+		imageCache:    s.mockCache,
+		registryStore: s.mockRegistryStore,
+		localScan:     s.mockLocalScan,
+		centralReady:  concurrency.NewSignal(),
+		centralClient: s.mockCentral,
+	}
+}
+
+func (s *imageServiceSuite) TestGetImage() {
+	ctx := context.Background()
+	imageID := "imageID"
+	imageName := "imageName"
+	err := errors.New("some error")
+	errCentral := errors.Wrap(err, "scanning image via central")
```

Comment:
In case you just want to check if the error contains a string, you can use `.ErrorContains` from testify. You can pass a string to match the error message instead of trying to match for an error value.

Reply:
I think I'll keep the errors. I'm currently using them in the return values of some of the `EXPECT` functions and if I change these to strings I'll have to create the errors there.